clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -12335,6 +12335,10 @@ def warn_hlsl_availability_unavailable :
   Warning<err_unavailable.Summary>,
   InGroup<HLSLAvailability>, DefaultError;
 
+def err_hlsl_intangible_type_cannot_be_declared : Error<
+"HLSL intangible type cannot be declared here">;
+def err_hlsl_intangible_type_as_function_arg_or_return : Error<
+"HLSL intangible type cannot be used as function %select{argument|return value}0">;
```

Comment:
I don't think this is correct. We do allow resources as argument and return types.